projects/RabbitMQ.Client/client/impl/RabbitMQActivitySource.cs

```diff
@@ -0,0 +1,232 @@
+﻿using System.Collections.Generic;
+using System.Diagnostics;
+using System.Reflection;
+using System.Text;
+using RabbitMQ.Client.Events;
+
+namespace RabbitMQ.Client
+{
+    public static class RabbitMQActivitySource
+    {
+        // These constants are defined in the OpenTelemetry specification:
+        // https://opentelemetry.io/docs/specs/semconv/messaging/messaging-spans/#messaging-attributes
+        internal const string MessageId = "messaging.message.id";
+        internal const string MessageConversationId = "messaging.message.conversation_id";
+        internal const string MessagingOperation = "messaging.operation";
+        internal const string MessagingSystem = "messaging.system";
+        internal const string MessagingDestination = "messaging.destination.name";
+        internal const string MessagingDestinationRoutingKey = "messaging.rabbitmq.destination.routing_key";
+        internal const string MessagingBodySize = "messaging.message.body.size";
+        internal const string MessagingEnvelopeSize = "messaging.message.envelope.size";
+        internal const string ProtocolName = "network.protocol.name";
+        internal const string ProtocolVersion = "network.protocol.version";
+        internal const string RabbitMQDeliveryTag = "messaging.rabbitmq.delivery_tag";
+
+        internal static readonly ActivitySource s_source = new ActivitySource(Name,
+            typeof(RabbitMQActivitySource).Assembly.GetCustomAttribute<AssemblyInformationalVersionAttribute>()
+                ?.InformationalVersion ?? "");
+
+        public const string Name = "RabbitMQ.Client";
+        public static bool UseRoutingKeyAsOperationName { get; set; } = true;
+        internal static bool HasListeners => s_source.HasListeners();
+
+        internal static readonly IEnumerable<KeyValuePair<string, object>> CreationTags = new[]
+        {
+            new KeyValuePair<string, object>(MessagingSystem, "rabbitmq"),
+            new KeyValuePair<string, object>(ProtocolName, "amqp"),
```

Comment:
nit: is it possible for RabbitMQ to ever use other protocol? If not, I suggest not populating it - created https://github.com/open-telemetry/semantic-conventions/issues/504

Reply:
What do you say @michaelklishin and @lukebakken?

Reply:
@lmolkova Apparently other protocols are supported via. plugins (https://www.rabbitmq.com/protocols.html) so I'm keeping this in here for the time being.